test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -0,0 +1,406 @@
+			}
+		}
+
+		return nil
+	}
+}
+
+// verifyUniqueLabels returns a verification function that checks if the label values for a given key are unique.
+func verifyUniqueLabels(key string) func(t *testing.T, tCtx ktesting.TContext, op realOp) error {
+	return func(t *testing.T, tCtx ktesting.TContext, op realOp) error {
+		labelValues, err := objLabelValues(t, tCtx, op, key)
+		if err != nil {
+			return fmt.Errorf("failed to get label values: %w", err)
+		}
+
+		seen := make(map[string]bool)
+		for _, labelValue := range labelValues {
+			if seen[labelValue] {
+				return fmt.Errorf("Node has duplicate label value %s for key %s", labelValue, key)
+			}
+			seen[labelValue] = true
+		}
+
+		return nil
+	}
+}
+
+// objLabelValues is a helper function to extract label values from the listed objects.
+// The listed objects are dependent on the operation type.
+func objLabelValues(t *testing.T, tCtx ktesting.TContext, op realOp, key string) ([]string, error) {
+	t.Helper()
+
+	labelValues := []string{}
+	switch op.(type) {
+	case *createNodesOp:
+		nodes, err := tCtx.Client().CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to list nodes for label verification: %w", err)
+		}
+
+		for _, node := range nodes.Items {
+			if labelValue, exists := node.Labels[key]; exists {
+				labelValues = append(labelValues, labelValue)
+			} else {
+				return nil, fmt.Errorf("Node %s is missing expected label %s", node.Name, key)
+			}
+		}
+	default:
+		return nil, fmt.Errorf("verifyLabel doesn't support this operation type: %T", op)
+	}
+
+	return labelValues, nil
+}
+
+// verifyObj checks if listed objects match the expected template object using cmp.Diff.
+func verifyObj(expectedObj any) func(t *testing.T, tCtx ktesting.TContext, op realOp) error {
+	return func(t *testing.T, tCtx ktesting.TContext, op realOp) error {
+		var cmpOpts []cmp.Option
+		var got, want any
+		switch opDetails := op.(type) {
+		case *createNodesOp:
+			var gotNodes []v1.Node
+			var expectedNodeTemplate *v1.Node
+
+			nodesList, err := tCtx.Client().CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to list nodes: %w", err)
+			}
+			gotNodes = nodesList.Items
+
+			var ok bool
+			expectedNodeTemplate, ok = expectedObj.(*v1.Node)
+			if !ok {
+				return fmt.Errorf("expectedObj must be *v1.Node when op is *createNodesOp, got %T", expectedObj)
+			}
+
+			wantNodes := make([]v1.Node, len(gotNodes))
+			// we don't need to verify len(), we just need to check if all of them are the same as the expected one.
+			for i := range gotNodes {
+				wantNodes[i] = *expectedNodeTemplate
+			}
+
+			cmpOpts = []cmp.Option{
+				cmpopts.EquateEmpty(),
+				cmpopts.IgnoreFields(metav1.ObjectMeta{},
+					"UID", "ResourceVersion", "Generation", "CreationTimestamp", "ManagedFields", "SelfLink", "Name",
+					"Labels", // verifyObj doesn't care about labels.
+				),
```

Comment:
Actually, can we ignore the entire ObjectMeta at v1.Node?

Reply:
We can't because GenerateNam should be checked in the customer template test.